src/Main.java

```diff
@@ -1,17 +1,82 @@
+import java.time.LocalDate;
+import java.util.Locale;
+
 // Press Shift twice to open the Search Everywhere dialog and type `show whitespaces`,
 // then press Enter. You can now see whitespace characters in your code.
 public class Main {
     public static void main(String[] args) {
-        // Press Alt+Enter with your caret at the highlighted text to see how
-        // IntelliJ IDEA suggests fixing it.
-        System.out.printf("Hello and welcome!");
+        zd1();
+        zd2();
+        zd3();
+    }
+    public static void zd1(){
+        int year = 2000;
+        leapYear(year);
 
-        // Press Shift+F10 or click the green arrow button in the gutter to run the code.
-        for (int i = 1; i <= 5; i++) {
 
-            // Press Shift+F9 to start debugging your code. We have set one breakpoint
-            // for you, but you can always add more by pressing Ctrl+F8.
-            System.out.println("i = " + i);
+
+    }
+    public static void leapYear(int year){
+
+        if (year%10 == 4 | year%10 == 8 | year%10 == 6 | year%10 == 2){
+            System.out.println(year+" год — високосный год ");
+
+        }
+        else if ( year%400 == 0){
+            System.out.println(year+" год — високосный год ");
         }
+            else {
+                System.out.println(year+" год — невисокосный год ");
+            }
+
+    }
+    public static void zd2(){
+        int type = 1;
+        int year = 1223;
+        String clientOC = typeOC(type);
+        String dataPhone = version(year);
+        System.out.println("Установите "+ dataPhone+" версию приложения для "+clientOC+" по ссылке");
+
     }
+    public static String typeOC(int type){
+        String name = "";
+        if (type == 0){
+            name = "iOS";
+
+        }
+        else{
+            name = "Android";
+        }
+        return name;
```

Comment:
Можно укоротить запись:
```suggestion
        if (type == 0){
            return "iOS";
        } else{
            return "Android";
        }
```